parsers/CO.py

```diff
@@ -1,45 +1,246 @@
+    "BAGAZO": "biomass",
+    "RAD SOLAR": "solar",
+    "VIENTO": "wind",
+    "COMBUSTOLEO": "oil",
+    "ACPM": "oil",
+    "JET-A1": "oil",
+}
+
+XM_DELAY = 2
 
+
+@refetch_frequency(timedelta(days=1))
 def fetch_consumption(
     zone_key: str = "CO",
     session: Session = Session(),
     target_datetime: Optional[datetime] = None,
     logger: Logger = getLogger(__name__),
-):
-    if target_datetime:
-        raise NotImplementedError("This parser is not yet able to parse past dates.")
-    with session.get(colombia_demand_URL, verify=False) as response:
-        demand_json = json.loads(response.content)
-        demand_data = demand_json["Variables"][0]["Datos"]
-
-        for datapoint in demand_data:
+) -> List[Dict[str, Any]]:
+    if target_datetime is None:
+        return fetch_live_consumption(zone_key, session, target_datetime, logger)
+    else:
+        return fetch_historical_consumption(zone_key, session, target_datetime, logger)
+
+
+def fetch_live_consumption(
+    zone_key: str = "CO",
+    session: Session = Session(),
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+) -> List[Dict[str, Any]]:
+    demand_list = TotalConsumptionList(logger)
+    response = session.get(colombia_demand_URL, verify=False)
+
+    demand_json = json.loads(response.content)
+    demand_data = demand_json["Variables"][0]["Datos"]
+
+    for datapoint in demand_data:
+        demand_list.append(
+            zoneKey=ZoneKey(zone_key),
+            datetime=arrow.get(datapoint["Fecha"], tzinfo=TZ).datetime,
+            consumption=round(datapoint["Valor"], 1),
+            source="xm.com.co",
+        )
+    return demand_list.to_list()
+
+
+def fetch_historical_consumption(
+    zone_key: str = "CO",
+    session: Session = Session(),
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+) -> List[Dict[str, Any]]:
+    demand_list = TotalConsumptionList(logger)
+    # Convert datetime to local time
+    target_arrow_in_tz = arrow.get(target_datetime).to(TZ)
+
+    objetoAPI = pydataxm.ReadDB()
+
+    # API request consumption
+    df_consumption = objetoAPI.request_data(
+        "DemaReal", "Sistema", target_arrow_in_tz.date(), target_arrow_in_tz.date()
+    )
+
+    if not df_consumption.empty:
+        hour_columns = [col for col in df_consumption.columns if "Hour" in col]
+        for hour_col in hour_columns:
+
+            target_datetime_in_tz = target_arrow_in_tz.datetime.replace(
+                hour=int(hour_col[-2:]) - 1
+            )
+            consumption = float(df_consumption[hour_col]) / 1000
             demand_list.append(
-                {
-                    "zoneKey": zone_key,
-                    "datetime": arrow.get(datapoint["Fecha"], tzinfo=TZ).datetime,
-                    "consumption": round(datapoint["Valor"], 1),
-                    "source": "xm.com.co",
-                }
+                zoneKey=ZoneKey(zone_key),
```

Comment:
```suggestion
                zoneKey=zone_key,
```
If the above is accepted, you don't need to cast it here.